usaspending_api/broker/management/commands/fabs_nightly_loader.py

```diff
@@ -14,41 +17,59 @@
 
+
 logger = logging.getLogger("script")
 
 
-SUBMISSION_LOOKBACK_MINUTES = 15
+LAST_LOAD_LOOKBACK_MINUTES = 15
+UPDATED_AT_MODIFIER_MS = 1
 
 
-def get_last_load_date():
+def get_incremental_load_start_datetime():
     """
-    Wraps the get_load_load_date helper which is responsible for grabbing the
-    last load date from the database.
-
-    Without getting into too much detail, SUBMISSION_LOOKBACK_MINUTES is used
-    to counter a very rare race condition where database commits are saved ever
-    so slightly out of order with the updated_at timestamp.  It will be
-    subtracted from the last_run_date to ensure submissions with similar
-    updated_at times do not fall through the cracks.  An unfortunate side
-    effect is that some submissions may be processed more than once which
-    SHOULDN'T cause any problems but will add to the run time.  To minimize
-    this, keep the value as small as possible while still preventing skips.  To
-    be clear, the original fabs loader did this as well, just in a way that did
-    not always prevent skips.
+    This function is designed to help prevent two issues we've discovered with the FABS nightly
+    pipline:
+
+     #1 LAST_LOAD_LOOKBACK_MINUTES are subtracted from last load datetime to counter a very rare
+        race condition where database commits are saved ever so slightly out of order when compared
+        to the last updated timestamp due to commit duration which can cause transactions to be
+        skipped.  The timestamp is set to when the commit begins, so if the commit starts before
+        the load but doesn't finish until after the load, the transactions saved as part of that
+        commit will never get picked up.  This has happened in the wild at least once and is not
+        all that hard to imagine if you consider that large submissions can take many minutes to
+        commit.  We do not currently have a good way to test this, but I've seen extremely large
+        transactions take an hour to commit.  I do not believe submissions currently get this large
+        but it's something to keep in mind.
+
+     #2 We use the minimum of the last load date or the max transaction_fabs updated_at date
+        to prevent FABS transactions submitted between when the source records are copied from
+        Broker and when FABS transactions are processed from being skipped.
+
+    An unfortunate side effect of the lookback is that some submissions may be processed more than
+    once.  This SHOULDN'T cause any problems since the FABS loader is designed to be able to reload
+    transactions, but it could add to the run time.  To minimize reprocessing, keep the
+    LAST_LOAD_LOOKBACK_MINUTES value as small as possible while still preventing skips.  To be
+    clear, the original fabs loader did this as well, just in a way that did not always prevent
+    skips (by always running since midnight - which had its own issues).
```

Comment:
I'd trim the last sentence. What the old loader was doing isn't important to someone looking at the new loader.